apps/src/templates/projects/SectionProjectsList.jsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import ProjectsList from './ProjectsList';
+import _ from 'lodash';
+import color from "../../util/color";
+
+const ALL_STUDENTS = '_all_students';
+
+const styles = {
+  filterDropdown: {
+    margin: 0,
+    color: 'dimgray'
+  },
+  filterRow: {
+    backgroundColor: color.teal,
+    borderBottom: 'solid 1px white',
+    color: color.white,
+    padding: 10,
+    fontSize: 14
+  },
+  filterSpan: {
+    float: 'right',
+    display: 'inline-flex',
+    alignItems: 'center',
+  },
+  clearDiv: {
+    clear: 'both'
+  }
+};
+
+class SectionProjectsList extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const studentNames = this.getStudentNames(props.projectsData);
+
+    this.state = {
+      studentNames,
+      selectedStudent: ALL_STUDENTS,
```

Comment:
I got this from these two sources:
* https://toddmotto.com/react-create-class-versus-component/ 
* https://facebook.github.io/react/docs/state-and-lifecycle.html#adding-local-state-to-a-class

technically it's not an implicit `setState` because it doesn't do all the things that `setState` does like potentially trigger re-rendering, rather it is the convention for setting initial state which replaces `getInitialState`.